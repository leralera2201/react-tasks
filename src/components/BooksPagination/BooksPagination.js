import React from "react";
import { Pagination, PaginationItem, PaginationLink } from "reactstrap";

import './BooksPagination.scss';

const BooksPagination = ({ pages, onPaginate, currentPage }) => {
  const paginatePrev = () => {
    if (currentPage !== 1) {
        onPaginate(currentPage - 1);
    }
  };

  const paginateNext = () => {
    if (currentPage !== pages.length) {
        onPaginate(currentPage + 1);
    }
  };

  return (
    <Pagination>
      <PaginationItem disabled={currentPage === 1} onClick={paginatePrev}>
        <PaginationLink previous />
      </PaginationItem>
      {pages.map((page) => (
        <PaginationItem key={page} onClick={() => onPaginate(page)}>
          <PaginationLink className={currentPage === page ? "active-page" : ""}>
            {page}
          </PaginationLink>
        </PaginationItem>
      ))}
      <PaginationItem
        disabled={currentPage === pages.length}
        onClick={paginateNext}
      >
        <PaginationLink next />
      </PaginationItem>
    </Pagination>
  );
};

export default BooksPagination;
