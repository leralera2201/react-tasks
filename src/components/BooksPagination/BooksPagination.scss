.active-page {
  background: #b0c4de !important;
}
