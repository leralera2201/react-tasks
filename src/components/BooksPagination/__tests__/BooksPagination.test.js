import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BooksPagination from '../BooksPagination';

describe('Testing BooksPagination component', () => {
  const props = {
    onPaginate: jest.fn(),
    pages: [1, 2, 3],
    currentPage: 1,
  };

  it('renders correctly', () => {
    const wrapper = shallow(<BooksPagination {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('BooksPagination prev will not render', () => {
    const wrapper = mount(<BooksPagination {...props} />);
    wrapper.find('#paginate-prev').first().simulate('click');
    expect(props.onPaginate).toHaveBeenCalledTimes(0);
  });

  it('BooksPagination prev will render', () => {
    const wrapper = mount(<BooksPagination {...props} />);
    wrapper.setProps({ currentPage: 2 });
    wrapper.find('#paginate-prev').first().simulate('click');
    expect(props.onPaginate).toHaveBeenCalledTimes(1);
  });

  it('BooksPagination next will render', () => {
    const wrapper = mount(<BooksPagination {...props} />);
    wrapper.find('#paginate-next').first().simulate('click');
    expect(props.onPaginate).toHaveBeenCalledTimes(1);
  });

  it('BooksPagination next will not render', () => {
    const wrapper = mount(<BooksPagination {...props} />);
    wrapper.setProps({ currentPage: 3 });
    wrapper.find('#paginate-next').first().simulate('click');
    expect(props.onPaginate).toHaveBeenCalledTimes(0);
  });
});
