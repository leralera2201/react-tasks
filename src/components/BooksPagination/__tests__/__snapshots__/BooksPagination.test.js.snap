// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BooksPagination component renders correctly 1`] = `
<Pagination
  aria-label='pagination'
  listTag='ul'
  tag='nav'
>
  <PaginationItem
    disabled={true}
    id='paginate-prev'
    onClick={[Function]}
    tag='li'
  >
    <PaginationLink
      previous={true}
      tag='a'
    />
  </PaginationItem>
  <PaginationItem
    key='1'
    onClick={[Function]}
    tag='li'
  >
    <PaginationLink
      className='active-page'
      tag='a'
    >
      1
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    key='2'
    onClick={[Function]}
    tag='li'
  >
    <PaginationLink
      className=''
      tag='a'
    >
      2
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    key='3'
    onClick={[Function]}
    tag='li'
  >
    <PaginationLink
      className=''
      tag='a'
    >
      3
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    disabled={false}
    id='paginate-next'
    onClick={[Function]}
    tag='li'
  >
    <PaginationLink
      next={true}
      tag='a'
    />
  </PaginationItem>
</Pagination>
`;
